src/index.js

```diff
@@ -0,0 +1,311 @@
+    core.setSecret(inputs.accessToken);
+    if (inputs.workspaceId) core.setSecret(inputs.workspaceId);
+    
+    // Log input configuration
+    logMessage(logFile, `Pipeline: ${inputs.pipeline}`);
+    logMessage(logFile, `API Endpoint: ${inputs.apiEndpoint}`);
+    logMessage(logFile, `Workspace ID: ${inputs.workspaceId || '<not set>'}`);
+    logMessage(logFile, `Compute Environment: ${inputs.computeEnv || '<not set>'}`);
+    
+    // Debug mode logging
+    if (inputs.debug) {
+      core.info('🐛 Debug mode enabled');
+      logMessage(logFile, 'Debug mode enabled', 'DEBUG');
+      core.info(`Pipeline: ${inputs.pipeline}`);
+      core.info(`API Endpoint: ${inputs.apiEndpoint}`);
+      core.info(`Workspace ID: ${inputs.workspaceId || '<not set>'}`);
+      core.info(`Compute Environment: ${inputs.computeEnv || '<not set>'}`);
+      core.info(`Revision: ${inputs.revision || '<not set>'}`);
+      core.info(`Work Directory: ${inputs.workdir || '<not set>'}`);
+      core.info(`Config Profiles: ${inputs.profiles || '<not set>'}`);
+      core.info(`Run Name: ${inputs.runName || '<not set>'}`);
+      core.info(`Labels: ${inputs.labels || '<not set>'}`);
+      core.info(`Wait: ${inputs.wait}`);
+      core.info(`Parameters: ${inputs.parameters ? `${inputs.parameters.length} chars` : '<not set>'}`);
+      core.info(`Nextflow Config: ${inputs.nextflowConfig ? `${inputs.nextflowConfig.length} chars` : '<not set>'}`);
+      core.info(`Pre-run Script: ${inputs.preRunScript ? `${inputs.preRunScript.length} chars` : '<not set>'}`);
+    }
+    
+    // Validate required inputs
+    if (!inputs.accessToken || inputs.accessToken.trim() === '') {
+      throw new Error(`access_token is required and cannot be empty.
+
+💡 Common causes:
+   • Missing TOWER_ACCESS_TOKEN secret in repository settings
+   • Secret value is empty or contains only whitespace
+   • Incorrect secret name in workflow file
+   
+📖 To fix this:
+   1. Go to your repository Settings → Secrets and variables → Actions
+   2. Add/update TOWER_ACCESS_TOKEN with your Seqera Platform token
+   3. Ensure your workflow uses: access_token: \${{ secrets.TOWER_ACCESS_TOKEN }}`);
+    }
+    
+    if (!inputs.pipeline) {
+      throw new Error('pipeline is required');
+    }
+    
+    // Initialize API client
+    const apiClient = new SeqeraPlatformAPI({
+      baseUrl: inputs.apiEndpoint,
+      accessToken: inputs.accessToken,
+      debug: inputs.debug
+    });
+    
+    // Test API connectivity
+    core.info('🔗 Testing API connectivity...');
+    logMessage(logFile, 'Testing API connectivity...');
+    const connectionTest = await apiClient.testConnection();
+    if (!connectionTest.success) {
+      logMessage(logFile, `API connectivity test failed: ${connectionTest.error}`, 'ERROR');
+      throw new Error(`API connectivity test failed: ${connectionTest.error}`);
+    }
+    core.info('✅ API connectivity confirmed');
+    logMessage(logFile, 'API connectivity confirmed');
```

Comment:
Is it possible to have only a single logger that goes to both? Feels a bit verbose / brittle to have to duplicate every log message..